DFE-Digital/early-careers-framework
# [CPDLP-1123] New providers can see old provider declarations

## Ticket and context

Ticket: https://dfedigital.atlassian.net/browse/CPDLP-1123

## Tech review

- New providers can see declarations made by old providers
- These are limited to certain states therefore excluding anything that is `voided` and `ineligible`
- Extracted query logic to a class which can be tested in isolation without the need to setup controller
- Patched ActiveRecord so it is capable of performing rudimentary `UNION` queries

### Is there anything that the code reviewer should know?

### Code quality checks

### HTML Checks

### Gotchas

## Product review

### How can someone see it it review app?

- As the old provider make some declarations against a participant
- Change their provider
- As the new provider make some declarations against the participant
- Fetch declarations as old provider
- Should only see old provider declarations
- Fetch declarations as new provider
- Should only see all provider declarations minus older provider `voided` and `ineligible` declarations

## External API changes

- I guess to some small degree, release notes updated
